**The project aims to predict whether a passenger referred by an existing customer will book a flight or not, based on various features such as seat comfort, cabin service, travel class,food beverage , entertainment service, etc. The prediction model is developed using classification techniques in machine learning.The use of machine learning techniques allows for the development of a model that can learn from historical passenger and booking data and make accurate predictions on new data. The model can be used by airlines to target marketing campaigns towards potential passengers who are likely to book a flight based on a referral from an existing customer.**

In [1]:
#import the libraries
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns


### Loading the Dataset

In [2]:
#load dataset
airline_df=pd.read_excel('data_airline_reviews.xlsx')

### Dataset First Insights

In [3]:
#dataset first look
airline_df.head()

airline  overall               author   review_date  \
0               NaN      NaN                  NaN           NaN   
1  Turkish Airlines      7.0  Christopher Hackley  8th May 2019   
2               NaN      NaN                  NaN           NaN   
3  Turkish Airlines      2.0        Adriana Pisoi  7th May 2019   
4               NaN      NaN                  NaN           NaN   

                                     customer_review aircraft  traveller_type  \
0                                                NaN      NaN             NaN   
1  ✅ Trip Verified | London to Izmir via Istanb...      NaN        Business   
2                                                NaN      NaN             NaN   
3  ✅ Trip Verified | Istanbul to Bucharest. We ...      NaN  Family Leisure   
4                                                NaN      NaN             NaN   

           cabin                         route           date_flown  \
0            NaN                           NaN                  NaN   
1  Economy Class  London to Izmir via Istanbul  2019-05-01 00:00:00   
2            NaN                           NaN                  NaN   
3  Economy Class         Istanbul to Bucharest  2019-05-01 00:00:00   
4            NaN                           NaN                  NaN   

   seat_comfort  cabin_service  food_bev  entertainment  ground_service  \
0           NaN            NaN       NaN            NaN             NaN   
1           4.0            5.0       4.0            4.0             2.0   
2           NaN            NaN       NaN            NaN             NaN   
3           4.0            1.0       1.0            1.0             1.0   
4           NaN            NaN       NaN            NaN             NaN   

   value_for_money recommended  
0              NaN         NaN  
1              4.0         yes  
2              NaN         NaN  
3              1.0          no  
4              NaN         NaN

In [4]:
#dataset rows and column count
print(airline_df.shape)

(131895, 17)


Total 131895 rows and 17 columns are present in our dataset

In [5]:
#dataset information
airline_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 131895 entries, 0 to 131894
Data columns (total 17 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   airline          65947 non-null  object 
 1   overall          64017 non-null  float64
 2   author           65947 non-null  object 
 3   review_date      65947 non-null  object 
 4   customer_review  65947 non-null  object 
 5   aircraft         19718 non-null  object 
 6   traveller_type   39755 non-null  object 
 7   cabin            63303 non-null  object 
 8   route            39726 non-null  object 
 9   date_flown       39633 non-null  object 
 10  seat_comfort     60681 non-null  float64
 11  cabin_service    60715 non-null  float64
 12  food_bev         52608 non-null  float64
 13  entertainment    44193 non-null  float64
 14  ground_service   39358 non-null  float64
 15  value_for_money  63975 non-null  float64
 16  recommended      64440 non-null  object 
dtypes: float64

### Checking Duplicates in Data

In [6]:
#checking for dupliacte value counts
airline_df.duplicated().value_counts()

True     70711
False    61184
Name: count, dtype: int64

In [7]:
#getting duplicate values
print(f'duplicate data count = {airline_df[airline_df.duplicated()].shape[0]}')
airline_df[airline_df.duplicated()]

duplicate data count = 70711


airline  overall author review_date customer_review aircraft  \
2          NaN      NaN    NaN         NaN             NaN      NaN   
4          NaN      NaN    NaN         NaN             NaN      NaN   
6          NaN      NaN    NaN         NaN             NaN      NaN   
8          NaN      NaN    NaN         NaN             NaN      NaN   
10         NaN      NaN    NaN         NaN             NaN      NaN   
...        ...      ...    ...         ...             ...      ...   
131885     NaN      NaN    NaN         NaN             NaN      NaN   
131887     NaN      NaN    NaN         NaN             NaN      NaN   
131889     NaN      NaN    NaN         NaN             NaN      NaN   
131891     NaN      NaN    NaN         NaN             NaN      NaN   
131893     NaN      NaN    NaN         NaN             NaN      NaN   

       traveller_type cabin route date_flown  seat_comfort  cabin_service  \
2                 NaN   NaN   NaN        NaN           NaN            NaN   
4                 NaN   NaN   NaN        NaN           NaN            NaN   
6                 NaN   NaN   NaN        NaN           NaN            NaN   
8                 NaN   NaN   NaN        NaN           NaN            NaN   
10                NaN   NaN   NaN        NaN           NaN            NaN   
...               ...   ...   ...        ...           ...            ...   
131885            NaN   NaN   NaN        NaN           NaN            NaN   
131887            NaN   NaN   NaN        NaN           NaN            NaN   
131889            NaN   NaN   NaN        NaN           NaN            NaN   
131891            NaN   NaN   NaN        NaN           NaN            NaN   
131893            NaN   NaN   NaN        NaN           NaN            NaN   

        food_bev  entertainment  ground_service  value_for_money recommended  
2            NaN            NaN             NaN              NaN         NaN  
4            NaN            NaN             NaN              NaN         NaN  
6            NaN            NaN             NaN              NaN         NaN  
8            NaN            NaN             NaN              NaN         NaN  
10           NaN            NaN             NaN              NaN         NaN  
...          ...            ...             ...              ...         ...  
131885       NaN            NaN             NaN              NaN         NaN  
131887       NaN            NaN             NaN              NaN         NaN  
131889       NaN            NaN             NaN              NaN         NaN  
131891       NaN            NaN             NaN              NaN         NaN  
131893       NaN            NaN             NaN              NaN         NaN  

[70711 rows x 17 columns]

In [8]:
#droping the duplicate values
airline_df.drop_duplicates(inplace=True)

In [9]:
#count duplicate values
airline_df.duplicated().sum()

np.int64(0)

### Handling Null Values

In [11]:
#count missing values/null values
print(airline_df.isnull().sum())

airline                1
overall             1783
author                 1
review_date            1
customer_review        1
aircraft           42696
traveller_type     23644
cabin               2479
route              23671
date_flown         23750
seat_comfort        4973
cabin_service       4944
food_bev           12843
entertainment      20954
ground_service     24015
value_for_money     1857
recommended         1423
dtype: int64


In [ ]:
#visualization of missing values
